action/protocol/account/transfer.go

```diff
@@ -63,19 +65,29 @@ func (p *Protocol) handleTransfer(ctx context.Context, act action.Action, sm pro
 		}
 	}
 
-	recipientAddr, err := address.FromString(tsf.Recipient())
+	var recipientAddr address.Address
+	if fCtx.CheckLegacyAddress {
+		recipientAddr, err = address.FromStringLegacy(tsf.Recipient())
+	} else {
+		recipientAddr, err = address.FromString(tsf.Recipient())
+	}
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed to decode recipient address %s", tsf.Recipient())
 	}
 	recipientAcct, err := accountutil.LoadAccount(sm, recipientAddr)
+	if !fCtx.CheckLegacyAddress {
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to load address %s", tsf.Recipient())
+		}
+	}
```

Comment:
this is a long-time unnoticed issue, in the case
```
err != nil && recipient is NOT a contract
```
the err escaped and will continue from line 109

after the HF, we can remove the `if !fCtx.CheckLegacyAddress` and always apply the check

Reply:
is it possible remove the height checking after the hardfork?